dbms/src/Storages/Page/V3/PageDirectory.cpp

```diff
@@ -1069,6 +1031,12 @@ void PageDirectory::apply(PageEntriesEdit && edit, const WriteLimiterPtr & write
     // stage 2, create entry version list for page_id.
     for (const auto & r : edit.getRecords())
     {
+        // Protected in write_lock
+        if (r.page_id.low > max_page_id)
```

Comment:
Can we use `max_applied_page_id` with some modifications of https://github.com/pingcap/tiflash/blob/2e8d324e76a2c5a96c83fdab8e7e08f3a5424d61/dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp#L114